### PR TITLE
[Graphbolt] Change probs to name of attribute

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
The original input is a tensor containing probs, this change it to a string point to an edge attribute.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/impl/csc_sampling_graph.py [hunk 1]
```diff
@@ -252,11 +252,12 @@ def sample_neighbors(
             Boolean indicating whether the edge IDs of sampled edges,
             represented as a 1D tensor, should be returned. This is
             typically used when edge features are required.
-        probs_or_mask: torch.Tensor, optional
-            Optional tensor containing the (unnormalized) probabilities
-            associated with each neighboring edge of a node. It must be a 1D
-            floating-point or boolean tensor with the number of elements equal
-            to the number of edges.
+        probs_name: str, optional
+            An optional string specifying the name of an edge attribute used a. This
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/csc_sampling_graph.py:256]:
```suggestion
            An optional string specifying the name of an edge attribute. This
```

File: python/dgl/graphbolt/impl/csc_sampling_graph.py [hunk 2]
```diff
@@ -302,7 +303,12 @@ def sample_neighbors(
             (fanouts >= 0) | (fanouts == -1)
         ), "Fanouts should consist of values that are either -1 or \
             greater than or equal to 0."
-        if probs_or_mask is not None:
+        if probs_name:
+            assert (
+                probs_name in self.edge_attributes
+            ), f"Unknown edge \
+                attribute '{probs_name}''."
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/csc_sampling_graph.py:310]:
```suggestion
            ), f"Unknown edge attribute '{probs_name}'."
```

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/impl/csc_sampling_graph.py (hunk 2 above)
Comment (peizhou001) [reply python/dgl/graphbolt/impl/csc_sampling_graph.py:310]: fixed

File: python/dgl/graphbolt/impl/csc_sampling_graph.py (hunk 1 above)
Comment (peizhou001) [reply python/dgl/graphbolt/impl/csc_sampling_graph.py]: fixed

**Pull request merged**